dotCMS/src/main/java/com/dotcms/webdav/FsDirectoryResource.java

```diff
@@ -0,0 +1,263 @@
+        }
+    }
+
+    @Override
+    public CollectionResource createCollection(String name) {
+        File fnew = new File(file, name);
+        boolean ok = fnew.mkdir();
+        if (!ok) {
+            throw new RuntimeException("Failed to create: " + fnew.getAbsolutePath());
+        }
+        return new FsDirectoryResource(host, factory, fnew, contentService);
+    }
+
+    @Override
+    public Resource child(String name) {
+        File fchild = new File(file, name);
+        return factory.resolveFile(this.host, fchild);
+
+    }
+
+    @Override
+    public List<? extends Resource> getChildren() {
+        ArrayList<FsResource> list = new ArrayList<FsResource>();
+        File[] files = this.file.listFiles();
+        if (files != null) {
+            for (File fchild : files) {
+                FsResource res = factory.resolveFile(this.host, fchild);
+                if (res != null) {
+                    list.add(res);
+                } else {
+                  Logger.info(FileSystemResourceFactory.class,"Couldnt resolve file " + fchild.getAbsolutePath());
+                }
+            }
+        }
+        return list;
+    }
+
+    /**
+     * Will redirect if a default page has been specified on the factory
+     *
+     * @param request
+     * @return
+     */
+    @Override
+    public String checkRedirect(Request request) {
+        if (factory.getDefaultPage() != null) {
+            return request.getAbsoluteUrl() + "/" + factory.getDefaultPage();
+        } else {
+            return null;
+        }
+    }
+
+    @Override
+    public Resource createNew(String name, InputStream in, Long length, String contentType) throws IOException {
+        File dest = new File(this.getFile(), name);
+        FileOutputStream out = null;
+        try {
+            out = new FileOutputStream(dest);
+            IOUtils.copy(in, out);
+        } finally {
+            IOUtils.closeQuietly(out);
+        }
+        // todo: ignores contentType
+        return factory.resolveFile(this.host, dest);
+
+    }
+
+    @Override
+    protected void doCopy(File dest) {
+        try {
+            FileUtils.copyDirectory(this.getFile(), dest);
+        } catch (IOException ex) {
+            throw new RuntimeException("Failed to copy to:" + dest.getAbsolutePath(), ex);
+        }
+    }
+
+    @Override
+    public LockToken createAndLock(String name, LockTimeout timeout, LockInfo lockInfo) throws NotAuthorizedException {
+        File dest = new File(this.getFile(), name);
+        createEmptyFile(dest);
+        FsFileResource newRes = new FsFileResource(host, factory, dest, contentService);
+        LockResult res = newRes.lock(timeout, lockInfo);
+        return res.getLockToken();
+    }
+
+    private void createEmptyFile(File file) {
+        FileOutputStream fout = null;
+        try {
+            fout = new FileOutputStream(file);
+        } catch (IOException e) {
+            throw new RuntimeException(e);

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)